### DISCUSSION TITLE
Caching

### DISCUSSION THREAD

**Discussion by allomorphy** [grafana/grafana-infinity-datasource#1006] [Ideas]
Hi all,  hi @yesoreyeram 

Thank you for this great work which we find very useful such that we are now using as standard to replace direct database queries in Grafana.
Our API is Fastapi and the database is Influxdb. All services are running on the same EC2 AWS instance.

Our endpoints are typically a fast database query and "rather simple" python logic. Our profiling shows that with th esimplest of cases with static return or small database query and not processing the result at all, the time cost is not much more than a direct db query from Grrafana.

However and sadly, we see on Query Inspection, a duration of the Grafana queries 300, 400, 500 or even more milliseconds.
This is mostly annoying in the UX when changing the time view frame in the time selector. Each move or zoom in the time view frame triggers a call to the API. These often accumulate and buffer up.

We do not tend to use queries that include the Grafana $timeFilter much at all. So there is no change in the query that the Infinity datasource sends to the api when we change the time selection. If the parameters passed to the api are unchanged, is this not a good case for caching of the return value from the Infinity datasource?

PS. I am assuming that this is not a currently available feature of this polugin but did notice the mention of caching on the following issue and am not sure if this is a feature which we could already access to address this use case.
https://github.com/grafana/grafana-infinity-datasource/issues/636#issue-1813090070

Thanks for shedding light on this point and thanks again for the great plugin.
E

**Comment by allomorphy** [comment]
This is  a non-issue as I see that caching in standard python addresses this well.

**Comment by yesoreyeram** [comment]
Caching is not implemented in the plugin level in favour of grafana enterprise query caching https://grafana.com/docs/grafana/latest/administration/data-source-management/#query-and-resource-caching
